docs/user/security/securing-communications/index.asciidoc

```diff
@@ -4,127 +4,178 @@
 
-Otherwise, if your certificate/key are in PEM format, specify them like so:
+If your PKCS#12 file is encrypted, add the decryption password to your <<secure-settings,{kib} keystore>>:
 
 [source,yaml]
 --------------------------------------------------------------------------------
-server.ssl.certificate: "/path/to/your/server.crt"
-server.ssl.key: "/path/to/your/server.key"
-server.ssl.keyPassphrase: "optional decryption password"
+bin/kibana-keystore add server.ssl.keystore.password
 --------------------------------------------------------------------------------
 
-After making these changes, you must always access {kib} via HTTPS. For example, 
-https://localhost:5601.
+TIP: If your PKCS#12 file isn't protected with a password, depending on how it was generated, you may need to set
+`server.ssl.keystore.password` to an empty string.
+--
+
++
+For more information about settings for certificates and keys, see <<settings,{kib} configuration settings>>.
 
-For more information, see <<settings,{kib} configuration settings>>.
+. *Configure {kib} to enable TLS for inbound connections.*
++
 --
+Specify that TLS/SSL is used in `kibana.yml`:
+
+[source,yaml]
+--------------------------------------------------------------------------------
+server.ssl.enabled: true
+--------------------------------------------------------------------------------
+--
+
+. *Restart {kib}.*
+
+After making these changes, you must always access {kib} via HTTPS. For example, https://<your_kibana_host>.com.
 
 [[configuring-tls-kib-es]]
 ==== Encrypting traffic between {kib} and {es}
 
-NOTE: To perform this step, you must 
-{ref}/configuring-security.html[enable the {es} {security-features}] or you 
-must have a proxy that provides an HTTPS endpoint for {es}. 
+NOTE: To perform this step, you must {ref}/configuring-security.html[enable the {es} {security-features}] or you must have a proxy that
+provides an HTTPS endpoint for {es}.
+
+. *{ref}/configuring-tls.html#tls-http[Enable TLS on the HTTP layer in {es}].*
 
-. Specify the HTTPS URL in the `elasticsearch.hosts` setting in the {kib}
-configuration file, `kibana.yml`:
+. *Obtain the certificate authority (CA) certificate chain for {es}.*
 +
 --
-[source,yaml]
+{kib} needs the appropriate CA certificate chain to properly establish trust when connecting to your {es} cluster.
+
+If you followed the {es} documentation for {ref}/configuring-tls.html#node-certificates[Generating Node Certificates], then you likely have
+a PKCS#12 file for each your {es} nodes. You can extract the CA certificate chain in PEM format from one of these files like so:
+
+[source,sh]
 --------------------------------------------------------------------------------
-elasticsearch.hosts: ["https://<your_elasticsearch_host>.com:9200"]
+openssl pkcs12 -in elastic-certificates.p12 -cacerts -nokeys -out elasticsearch-ca.crt
 --------------------------------------------------------------------------------
+--
```

Comment:
Second step, tell the user to obtain the CA certificate chain. If they followed the Elasticsearch instructions for generating Node certificates, they will wind up with one or more P12 files. If they generated a separate CA using the `elasticsearch-certutil ca` command, OR if they only have a single node and did not generate a separate CA, the CA cert can be obtained from the `elastic-certificates.p12` file.

Note: this doesn't cover the edge case if the user has multiple nodes and decided to generate multiple P12 files without using a CA (each node would have its own CA).